Errors with Upgrade to 0.68-preview

### Problem Description

After upgrading to 0.68.0-preivew.1 Gallery is hitting the following errors. 

Seems like the new way of adding CppWinrt with packagereference isn't playing well with ReactNativeXaml and that a series of other modules are unable to build because of something to do with the UAP framework. 
![image](https://user-images.githubusercontent.com/34109996/153282698-5beb821a-877f-41ae-803d-ac356082052e.png)


### Steps To Reproduce

Upgrade gallery to 0.68 preview.

### Expected Results

Successful build. 

### CLI version

7.0.1

### Environment

```markdown
info Fetching system and libraries information...
System:
    OS: Windows 10 10.0.22000
    CPU: (8) x64 Intel(R) Core(TM) i7-9700K CPU @ 3.60GHz
    Memory: 49.50 GB / 63.84 GB
  Binaries:
    Node: 16.13.2 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.5 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 8.1.2 - C:\Program Files\nodejs\npm.CMD
    Watchman: Not Found
  SDKs:
    Android SDK:
      API Levels: 28, 29, 30
      Build Tools: 28.0.3, 29.0.2, 29.0.3, 30.0.2, 30.0.3
      System Images: android-29 | Intel x86 Atom_64, android-29 | Google APIs Intel x86 Atom, android-29 | Google Play Intel x86 Atom, android-30 | Google APIs Intel x86 Atom
      Android NDK: Not Found
    Windows SDK:
      AllowDevelopmentWithoutDevLicense: Enabled
      AllowAllTrustedApps: Enabled
      Versions: 10.0.18362.0, 10.0.19041.0
  IDEs:
    Android Studio: Version  4.1.0.0 AI-201.8743.12.41.7199119
    Visual Studio: 16.9.32002.222 (Visual Studio Enterprise 2019)
  Languages:
    Java: 1.8.0_282
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.2 => 17.0.2
    react-native: 0.68.0-rc.1 => 0.68.0-rc.1
    react-native-windows: 0.68.0-preview.1 => 0.68.0-preview.1
  npmGlobalPackages:
    *react-native*: Not Found
```


### Target Platform Version

10.0.19041

### Target Device(s)

Desktop

### Visual Studio Version

_No response_

### Build Configuration

Debug

### Snack, code example, screenshot, or link to a repository

_No response_

This is an unexpected error.

There is a failure building even just Microsoft.ReactNative (meaning, it could be unrelated to the gallery code or dependencies):
```
msbuild.exe /p:Configuration=Debug /p:Platform=x64 /p:AppxBundle=Never /p:RestorePackagesConfig=true /p:RunAutolinkCheck=false /restore .\node_modules\react-native-windows\Microsoft.ReactNative\ /p:UseWinUI3=false
```

The two resulting errors are:

1. Assertion failed (apparently, `noexcept` method returning void).
```
node_modules\react-native-windows\Mso\dispatchQueue\dispatchQueue.h(728,17): error C2607: static assertion failed [R:\pkr\gal\node_modules\react-native-windows\Microsoft.ReactNative\Mi
crosoft.ReactNative.vcxproj
```

And, the one likely shown in VS:
```
(_OnXamlPreCompileError target) ->
  C:\Program Files (x86)\Windows Kits\10\bin\10.0.19041.0\XamlCompiler\Microsoft.Windows.UI.Xaml.Common.targets(486,5): error MSB4181: The "CompileXaml" task returned false but did not log an error
. [node_modules\react-native-windows\Microsoft.ReactNative\Microsoft.ReactNative.vcxproj]
```

The latter does not happen if we set MSBuild property `UseWinUI3` to `true`, but the `dispatchQueue` error still happens.

--- a/change/react-native-windows-fe9965b7-8e26-4928-b2d8-7453a5738923.json
+++ b/change/react-native-windows-fe9965b7-8e26-4928-b2d8-7453a5738923.json
@@ -1,0 +1,7 @@
+{
+  "type": "prerelease",
+  "comment": "Set ResolveNuGetPackages in external props",
+  "packageName": "react-native-windows",
+  "email": "julio.rocha@microsoft.com",
+  "dependentChangeType": "patch"
+}

--- a/vnext/PropertySheets/External/Microsoft.ReactNative.Common.props
+++ b/vnext/PropertySheets/External/Microsoft.ReactNative.Common.props
@@ -41,6 +41,9 @@
   <PropertyGroup Label="NuGet" Condition="'$(MSBuildProjectExtension)' == '.vcxproj'">
     <!--See https://docs.microsoft.com/en-us/nuget/reference/msbuild-targets#restore-target-->
     <RestoreUseStaticGraphEvaluation Condition="'$(BuildingInsideVisualStudio)' == 'true'">true</RestoreUseStaticGraphEvaluation>
+
+    <!-- Ensure PackageReference compatibility for any consuming projects/apps -->
+    <ResolveNuGetPackages>false</ResolveNuGetPackages>
   </PropertyGroup>
 
   <Import Project="$(ReactNativeWindowsDir)\PropertySheets\Generated\PackageVersion.g.props" />

--- a/vnext/PropertySheets/External/Microsoft.ReactNative.Uwp.CppLib.props
+++ b/vnext/PropertySheets/External/Microsoft.ReactNative.Uwp.CppLib.props
@@ -16,12 +16,6 @@
     <SkipAddPriPayloadFilesToCopyToOutputDirectoryItems>true</SkipAddPriPayloadFilesToCopyToOutputDirectoryItems>
     <GenerateLibraryLayout>false</GenerateLibraryLayout>
   </PropertyGroup>
-
-  <!-- Ensure PackageReference compatibility for any consuming projects/apps -->
-  <PropertyGroup Label="NuGet" Condition="'$(MSBuildProjectExtension)' == '.vcxproj'">
-    <ResolveNuGetPackages>false</ResolveNuGetPackages>
-  </PropertyGroup>
-
   <Import Project="$(MSBuildThisFileDirectory)Microsoft.ReactNative.Uwp.Common.props" />
   <Import Project="$(ReactNativeWindowsDir)\PropertySheets\WinUI.props" />
   <Import Project="$(ReactNativeWindowsDir)\PropertySheets\PackageVersionDefinitions.props" />

--- a/vnext/PropertySheets/External/Microsoft.ReactNative.Uwp.CppLib.props
+++ b/vnext/PropertySheets/External/Microsoft.ReactNative.Uwp.CppLib.props
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<!-- 
+<!--
   Copyright (c) Microsoft Corporation.
   Licensed under the MIT License.
 
@@ -16,6 +16,12 @@
     <SkipAddPriPayloadFilesToCopyToOutputDirectoryItems>true</SkipAddPriPayloadFilesToCopyToOutputDirectoryItems>
     <GenerateLibraryLayout>false</GenerateLibraryLayout>
   </PropertyGroup>
+
+  <!-- Ensure PackageReference compatibility for any consuming projects/apps -->
+  <PropertyGroup Label="NuGet" Condition="'$(MSBuildProjectExtension)' == '.vcxproj'">
+    <ResolveNuGetPackages>false</ResolveNuGetPackages>
+  </PropertyGroup>
+
   <Import Project="$(MSBuildThisFileDirectory)Microsoft.ReactNative.Uwp.Common.props" />
   <Import Project="$(ReactNativeWindowsDir)\PropertySheets\WinUI.props" />
   <Import Project="$(ReactNativeWindowsDir)\PropertySheets\PackageVersionDefinitions.props" />

